src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -142,12 +148,12 @@ export const useSwapper = () => {
   // This will instantiate a manager with no swappers
   // Swappers will be added in the useEffect below
   const [swapperManager, setSwapperManager] = useState<SwapperManager>(() => new SwapperManager())
-
+  const featureFlags = useSelector(selectFeatureFlags)
   useEffect(() => {
     ;(async () => {
       setSwapperManager(await getSwapperManager())
     })()
-  }, [])
+  }, [featureFlags])
```

Comment:
To quote our discussion on Discord:

> I can see why this works, as we get a new `useSwapper()` instance everytime we change trade routes, so you can't really the `useEffect()` hook in `useSwapper()` itself to react on a new reference

I don't think this does what it intends to i.e react on a new `featureFlags` reference to rerender the swapper.

You should be able to revert this hunk.

Reply:
yep it seems to work fine without it. done